full_snapshot.proto

```diff
@@ -508,3 +511,27 @@ message FunctionStatistic {
   double total_time = 3;
   double self_time = 4;
 }
+
+message CustomTypeInformation {
+  int32 database_idx = 1;
+  string schema_name = 2;
+  string name = 3;
+  Type type = 4;
+  string domain_type = 5;
+  bool domain_not_null = 6;
+  string domain_default = 7;
+  string domain_constraint = 8;
+  repeated string enum_values = 9;
+  repeated CompositeAttr composite_attrs = 10;
+
+  enum Type {
+    ENUM = 0;
+    DOMAIN = 1;
+    COMPOSITE = 2;
```

Comment:
As commented in https://github.com/pganalyze/collector/pull/196, we may want to add range/multirange/pseudo types here as well.